packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -126,6 +126,7 @@ class StackViewport extends Viewport implements IStackViewport {
   // TODO: These should not be here and will be nuked
   public modality: string; // this is needed for tools
   public scaling: Scaling;
+  private scalingCache: { [key: string]: ScalingParameters } = {};
```

Comment:
It looks like this will get bigger and bigger as the stack is reused. We should nuke its contents when setstack changes the imageIds hosted on the stack.


Also instead of `{ [key: string]: ScalingParameters } = {};`, its more natural to use Record here in TypeScript:

`private scaleCache: Record<string, ScalingParameters> = {}`

Reply:
yeah I thought about this after I logged off for the day, I will take care of it